api/infrastructure/parallelcluster-api.yaml

```diff
@@ -0,0 +1,246 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: 'Template for the ParallelCluster API'
+
+Parameters:
+  EnableMultiRegionSupport:
+    Description: When set to true it enables the API to control clusters in all regions
+    Type: String
+    Default: false
+    AllowedValues:
+      - true
+      - false
+
+  EcrImageUri:
+    Description: When specified use this image for the Lambda function and skip the import phase
+    Type: String
+    Default: ''
+
+  ApiDefinitionS3Uri:
+    Description: S3 URI of the ParallelCluster API spec
+    Type: String
+    Default: uri_of_pc_api # TODO change with actual URI
+
+  CustomDomainName:
+    Description: When specified, the custom domain name of the ParallelCluster API
+    Type: String
+    Default: ''
+
+  CustomDomainCertificate:
+    Description: When specified, the ARN of the certificate for custom domain name of the ParallelCluster API
+    Type: String
+    Default: ''
+
+  CustomDomainHostedZoneId:
+    Description: When specified, the id of the Hosted Zone of the custom domain name of the ParallelCluster API
+    Type: String
+    Default: ''
+
+  PublicEcrImageUri:
+    Description: When specified, the URI of the Docker image for the Lambda of the ParallelCluster API
+    Type: String
+    Default: uri_of_public_ecr_image # TODO change with actual URI
+
+  ParallelClusterVersion:
+    Description: The ParallelCluster version
+    Type: String
+    Default: 3.0.0
+
+
+# More info about Globals: https://github.com/awslabs/serverless-application-model/blob/master/docs/globals.rst
+Globals:
+  Function:
+    Timeout: 30
+    MemorySize: 256
+
+  Api:
+    Auth:
+      DefaultAuthorizer: AWS_IAM
+    TracingEnabled: True
+    EndpointConfiguration:
+      Type: REGIONAL
+
+
+Conditions:
+  UsesCustomEcrImageUri: !Not [!Equals [!Ref EcrImageUri, '']]
+  DoesNotUseCustomEcrImageUri: !Not [!Condition UsesCustomEcrImageUri]
+  UsesCustomDomain: !Not [!Equals [!Ref CustomDomainName, '']]
+  UsesRoute53Configuration: !Not [!Equals [!Ref CustomDomainHostedZoneId, '']]
+  UsesOnlyCustomDomain: !And
+    - !Condition UsesCustomDomain
+    - !Not [!Condition UsesRoute53Configuration]
+  UsesCustomDomainAndRoute53Configuration: !And
+    - !Condition UsesCustomDomain
+    - !Condition UsesRoute53Configuration
+  DoesNotUseCustomDomain: !Not [!Condition UsesCustomDomain]
+
+
+Resources:
+
+  # We need to define three AWS::Serverless::Api due to an issue with the handling of AWS::NoValue
+  # See related GitHub issue: https://github.com/aws/serverless-application-model/issues/1435
+
+  ApiGatewayApiWithOnlyCustomDomain:
+    Condition: UsesOnlyCustomDomain
+    Type: AWS::Serverless::Api
+    Properties:
+      Tags:
+        parallelcluster: !Ref ParallelClusterVersion
```

Comment:
let's use `parallelcluster:version` as the tag key. Also we might want to add this to the globals

Reply:
The `Tags` property can be specified in the `Globals` section for `AWS::Serverless::Function`, but not for `AWS::Serverless::Api` (see [documentation](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-specification-template-anatomy-globals.html)), I will change the name and put it in the `Globals` section for the `Function` resource type